pkg/cvc/v1alpha1/build.go

```diff
@@ -0,0 +1,282 @@
+// with the ones that are provided here
+func (b *Builder) WithAnnotations(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolumeclaim object: missing annotations"),
+		)
+		return b
+	}
+
+	if b.cvc.object.Annotations == nil {
+		return b.WithAnnotationsNew(annotations)
+	}
+
+	for key, value := range annotations {
+		b.cvc.object.Annotations[key] = value
+	}
+	return b
+}
+
+// WithAnnotationsNew resets existing annotations of CStorVolumeiClaim if any with
+// ones that are provided here
+func (b *Builder) WithAnnotationsNew(annotations map[string]string) *Builder {
+	if len(annotations) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolumeclaim object: no new annotations"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newannotations := map[string]string{}
+	for key, value := range annotations {
+		newannotations[key] = value
+	}
+
+	// override
+	b.cvc.object.Annotations = newannotations
+	return b
+}
+
+// WithLabels merges existing labels of CStorVolumeClaim if any
+// with the ones that are provided here
+func (b *Builder) WithLabels(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolumeclaim object: missing labels"),
+		)
+		return b
+	}
+
+	if b.cvc.object.Labels == nil {
+		return b.WithLabelsNew(labels)
+	}
+
+	for key, value := range labels {
+		b.cvc.object.Labels[key] = value
+	}
+	return b
+}
+
+// WithLabelsNew resets existing labels of CStorVolumeClaim if any with
+// ones that are provided here
+func (b *Builder) WithLabelsNew(labels map[string]string) *Builder {
+	if len(labels) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolumeclaim object: no new labels"),
+		)
+		return b
+	}
+
+	// copy of original map
+	newlbls := map[string]string{}
+	for key, value := range labels {
+		newlbls[key] = value
+	}
+
+	// override
+	b.cvc.object.Labels = newlbls
+	return b
+}
+
+// WithFinalizers merges existing finalizers of CStorVolumeClaim if any
+// with the ones that are provided here
+func (b *Builder) WithFinalizers(finalizers []string) *Builder {
+	if len(finalizers) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolumeclaim object: missing finalizers"),
+		)
+		return b
+	}
+
+	if b.cvc.object.Finalizers == nil {
+		return b.WithFinalizersNew(finalizers)
+	}
+
+	b.cvc.object.Finalizers = append(b.cvc.object.Finalizers, finalizers...)
+	return b
+}
+
+// WithFinalizersNew resets existing finalizers of CStorVolumeClaim if any with
+// ones that are provided here
+func (b *Builder) WithFinalizersNew(finalizers []string) *Builder {
+	if len(finalizers) == 0 {
+		b.errs = append(
+			b.errs,
+			errors.New("failed to build cstorvolumeclaim object: no new finalizers"),
+		)
+		return b
+	}
+
+	// override
+	b.cvc.object.Finalizers = finalizers
```

Comment:
Should we copy the array before assigning

Reply:
Is it really needed, since finalizers variable is not a pointer.